dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceServiceTest.java

```diff
@@ -342,4 +345,45 @@ void testTrackedEntityInstanceGridWithNoDisplayAttributes()
         assertThrows( IllegalArgumentException.class,
             () -> entityInstanceService.getTrackedEntityInstancesGrid( params ) );
     }
+
+    @Test
+    void shouldCountOneEntityWhenOneAdded()
+    {
+        injectSecurityContext( superUser );
+        filtH.setDisplayInListNoProgram( true );
+        attributeService.addTrackedEntityAttribute( filtH );
+
+        entityInstanceA1.setTrackedEntityType( trackedEntityTypeA );
+        entityInstanceService.addTrackedEntityInstance( entityInstanceA1 );
+
+        TrackedEntityAttributeValue trackedEntityAttributeValue = new TrackedEntityAttributeValue();
+        trackedEntityAttributeValue.setAttribute( filtH );
+        trackedEntityAttributeValue.setEntityInstance( entityInstanceA1 );
+        trackedEntityAttributeValue.setValue( ATTRIBUTE_VALUE );
+        attributeValueService.addTrackedEntityAttributeValue( trackedEntityAttributeValue );
```

Comment:
I know the existing tests don't make it easy 😬 Do you have some time to figure out how to make the setup easier to read? The "arrange" part of the test is pretty long and its hard to see what matters. Which of these need to match the params so that it will get the correct count...